src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -688,6 +803,24 @@ private static void RemoveElementAfterClearTag(XElement sectionElement)
             }
         }
 
+        /// <summary>
+        /// Checks if a section contains clear tag.
+        /// </summary>
+        /// <param name="section">XElement section.</param>
+        private static bool ContainsClearTag(XElement section)
+        {
+            if (section == null)
+            {
+                return false;
+            }
+
+            return section
+                .Nodes()
+                .Where(n => n.NodeType == XmlNodeType.Element)
+                .Select(n => (XElement)n)
+                .Any(e => string.Equals(e.Name.LocalName, "clear", StringComparison.OrdinalIgnoreCase));
```

Comment:
Suggestion:  create utility method(s) like:
```
private static bool IsClearTag(XElement element)
{
    return element.Name.LocalName.Equals("clear", StringComparison.OrdinalIgnoreCase);
}
```
...and replace all `string.Equals(...)` comparisons with it.  I count 4 "clear"-related string comparsions in this file before your PR.  You could do the same for "add".